### DISCUSSION TITLE
Question about convergence with decreasing residuals

### DISCUSSION THREAD

**Discussion by Bala-1005** [idaholab/moose#17322] [Q&A Modules: Phase field] [answered]
Hello everyone,

I have noticed a very weird convergence issue while running my phasefield-mechanics problem on moose. Attached are two parts of the same output file. 
Even though the linear and nonlinear residuals drop in a similar way, one of the time steps returns a message saying the solution has converged but the other returns saying solve has not converged. Why would it happen that way?

Thanks,
Bala
[converged.txt](https://github.com/idaholab/moose/files/6143871/converged.txt)
[nonconverged.txt](https://github.com/idaholab/moose/files/6143872/nonconverged.txt)

**Comment by jessecarterMOOSE** [comment, marked as answer]
Looks like you have the maximum number of nonlinear iterations (`nl_max_its`) set to 10. Try increasing that value. The other solve probably converged because the default nonlinear tolerance (`nl_rel_tol`) of 1e-8 was met. The nonconverged solution was just slightly above that value (comparing the last nonlinear residual to the first) as it hit the 10th iteration so it aborted.

**Comment by jessecarterMOOSE** [comment]
Check out this page:
https://mooseframework.inl.gov/application_usage/failed_solves.html

**Reply by Bala-1005** [reply]
Thank you @jessecarterMOOSE . That solved the issue. Just to understand better, when you say the nonlinear relative tolerance was not met in the non converged solution do you mean to say that the difference between the final nonlinear tolerance and the previous nonlinear tolerance?

Also, this may be unrelated but I see the letter 'm' in my output file between numbers when it displays the residuals? What does this mean and how do I remove this?

Thanks, 
Bala

**Reply by jessecarterMOOSE** [reply]
It is the ratio of the final nonlinear residual to the first, or rather the 0th, nonlinear residual of the particular step.

All that extra junk is color code information that a simple text editor doesn't understand. You can run moose with `--no-color` to disable color output.

**Reply by Bala-1005** [reply]
Thank you so much @jessecarterMOOSE .
